src/s1reader/s1_orbit.py

```diff
@@ -332,12 +398,278 @@ def get_orbit_file_from_list(zip_path: str, orbit_file_list: list) -> str:
+            break
+
+    # if 1. and 2. are successful, then try to concatenate them
+    if resorb_filename_earlier and resorb_filename_later:
+        if concatenate_resorb:
+            # BE CAREFUL ABOUT THE ORDER HOW THEY ARE CONCATENATED.
+            # See NOTE in retrieve_orbit_file() for detail.
+            concat_resorb_filename = combine_xml_orbit_elements(resorb_filename_later,
+                                                                resorb_filename_earlier)
+            print('RESORB Concatenation successful.')
+            return concat_resorb_filename
+        else:
+            return [resorb_filename_earlier, resorb_filename_later]
+
+    print('Cannot find a single RESORB file that meets the time frame criteria.')
+    return
+
+
+def combine_xml_orbit_elements(file1: str, file2: str) -> str:
+    """Combine the orbit elements from two XML files.
+
+    Create a new .EOF file with the combined results.
+    Output is named with the start_datetime and stop_datetime changed, with
+    the same base as `file1`.
+
+    Parameters
+    ----------
+    file1 : str
+        The path to the first .EOF file.
+    file2 : str
+        The path to the second .EOF file.
+
+    Returns
+    -------
+    _ : str
+        Name of the newly created EOF file.
+    """
+
+    def get_dt(root: ElementTree.ElementTree, tag_name: str) -> datetime.datetime:
+        time_str = root.find(f".//{tag_name}").text.split("=")[-1]
+        return datetime.datetime.fromisoformat(time_str)
+
+    # Parse the XML files
+    tree1 = ElementTree.parse(file1)
+    tree2 = ElementTree.parse(file2)
+
+    root1 = tree1.getroot()
+    root2 = tree2.getroot()
+
+    # Extract the Validity_Start and Validity_Stop timestamps from both files
+    start_time1 = get_dt(root1, "Validity_Start")
+    stop_time1 = get_dt(root1, "Validity_Stop")
+    start_time2 = get_dt(root2, "Validity_Start")
+    stop_time2 = get_dt(root2, "Validity_Stop")
+
+    # Determine the new Validity_Start and Validity_Stop values
+    new_start_dt = min(start_time1, start_time2)
+    new_stop_dt = max(stop_time1, stop_time2)
+
+    # Update the Validity_Start and Validity_Stop timestamps in the first XML
+    root1.find(".//Validity_Start").text = "UTC=" + new_start_dt.strftime(
+        "%Y-%m-%dT%H:%M:%S"
+    )
+    root1.find(".//Validity_Stop").text = "UTC=" + new_stop_dt.strftime(
+        "%Y-%m-%dT%H:%M:%S"
+    )
+
+    # Combine the <OSV> elements
+    list_of_osvs1 = root1.find(".//List_of_OSVs")
+    list_of_osvs2 = root2.find(".//List_of_OSVs")
+
+    merge_osv_list(list_of_osvs1, list_of_osvs2)
+
+    outfile = _generate_filename(file1, new_start_dt, new_stop_dt)
+    tree1.write(outfile, encoding="UTF-8", xml_declaration=True)
+    return outfile
+
+
+def merge_osv_list(list_of_osvs1, list_of_osvs2):
+    # Extract the UTC from the OSV of the first XML
+    osv1_utc_list = [datetime.datetime.fromisoformat(osv1.find('UTC').text.replace('UTC=',''))
```

Comment:
Let's make `datetime.datetime.fromisoformat(osv1.find('UTC').text.replace('UTC=','')` it's own function in `s1_orbit.py` because it's repeated several times in this PR. Maybe call it `strip_UTC_from_time_string`?

Reply:
Good idea. I've defined a function `_get_uct_time_from_osv()` that extracts UTC time from osv.